File: app/javascript/channels/updates_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("UpdatesChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    console.log('Live Updates Connected')
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
    console.log('Live Updates Disconnected')
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    console.log('Updating Data...')
  },

  char_level_up(char) {
    // level up char
    this.perform("char_level_up", char || null)
  },

  new_structure(structure) {
    this.perform("new_structure", structure)
  }
  
});

File: app/channels/updates_channel.rb
class UpdatesChannel < ApplicationCable::Channel

  def subscribed
    current_user.connected = true
    current_user.save
    stream_from "updates"

  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
    current_user.connected = false
    current_user.save
  end

end
